### PR TITLE
feat: add GPT-5.4 support and docs refresh

### PR DESCRIPTION
**Pull request by ndycode**
## Summary
This PR updates the plugin to support the GPT-5.4 release while keeping rollout safe for existing users.

- Adds runtime support for `gpt-5.4` as the new default general-purpose model family.
- Adds optional/manual support for `gpt-5.4-pro` (not enabled by default in shipped templates).
- Adds a default fallback edge in fallback mode: `gpt-5.4-pro -> gpt-5.4`.
- Keeps legacy fallback behavior unchanged (`gpt-5.3-codex -> gpt-5.2-codex` via existing compatibility toggle).
- Refreshes config templates and docs so examples/default usage now point to GPT-5.4.
- Expands automated tests to cover GPT-5.4 and GPT-5.4-pro normalization, reasoning handling, and fallback behavior.

### Runtime/model handling
- `lib/request/helpers/model-map.ts`
  - Added mappings for GPT-5.4 variants (`none`, `low`, `medium`, `high`, `xhigh`) -> `gpt-5.4`.
  - Added mappings for GPT-5.4 Pro variants (`low`, `medium`, `high`, `xhigh`) -> `gpt-5.4-pro`.
- `lib/request/request-transformer.ts`
  - Added normalize logic for `gpt-5.4` and `gpt-5.4-pro`.
  - Added reasoning-effort rules:
    - `gpt-5.4` supports `none` and `xhigh`.
    - `gpt-5.4-pro` supports `xhigh`; `none` is upgraded to `low`.
- `index.ts`
  - Updated quota probe model order to include `gpt-5.4` first.

### Fallback policy and entitlement behavior
- `lib/request/fetch-helpers.ts`
  - Added default fallback chain edge for fallback mode:
    - `gpt-5.4-pro -> gpt-5.4`
  - Updated entitlement guidance message to include `gpt-5.4` as latest general fallback option.

### Prompt family/instructions
- `lib/prompts/codex.ts`
  - Added `gpt-5.4` model family detection.
  - Added cache file handling for GPT-5.4 instructions.
  - Mapped GPT-5.4 to current upstream prompt family file (`gpt_5_2_prompt.md`) per current Codex prompt layout.
  - Included GPT-5.4 in instruction prewarm candidates.

### Templates/docs/versioning
- Config templates now default to GPT-5.4:
  - `config/opencode-modern.json`
  - `config/opencode-legacy.json`
- Documentation refreshed for GPT-5.4 defaults + optional/manual GPT-5.4 Pro usage:
  - `README.md`
  - `config/README.md`
  - `docs/getting-started.md`
  - `docs/configuration.md`
  - `docs/troubleshooting.md`
  - `docs/index.md`
  - `docs/development/TESTING.md`
- Version and changelog:
  - `package.json` -> `5.4.2`
  - `CHANGELOG.md` entry for `5.4.2`

## Test Coverage Added
- Unit/integration coverage for GPT-5.4 and GPT-5.4-pro in:
  - `test/model-map.test.ts`
  - `test/request-transformer.test.ts`
  - `test/fetch-helpers.test.ts`
  - `test/codex-prompts.test.ts`
  - `test/index.test.ts`
- Property-test input space expanded:
  - `test/property/helpers.ts`
  - `test/property/transformer.property.test.ts`

## Validation
- `npm run typecheck` passed
- `npm run lint` passed
- `npm test` passed
  - 56 test files
  - 1797 tests

## Notes for Reviewers
- `gpt-5.4-pro` remains opt-in/manual in templates to avoid entitlement surprises.
- Existing legacy fallback path behavior is preserved.
- PR scope intentionally includes runtime + tests + docs to keep model rollout consistent across code and user guidance.

## Summary by CodeRabbit

* **New Features**
  * Added support for GPT-5.4 and optional GPT-5.4-Pro models with normalization, reasoning behavior, and fallback handling.

* **Documentation**
  * Updated configuration, getting-started, troubleshooting, and readme content to reflect GPT-5.4/5.4‑Pro availability and fallback mappings.

* **Tests**
  * Expanded test coverage for normalization, reasoning configs, fallbacks, and extensive edge cases for the GPT-5.4 family.

* **Bug Fixes**
  * Quota probing now prioritizes GPT-5.4 before legacy Codex models.

* **Chores**
  * Version bumped to 5.4.2.

**note:** greptile review for oc-chatgpt-multi-auth. cite files like `lib/foo.ts:123`. confirm regression tests + windows concurrency/token redaction coverage.
---

<h3>Greptile Summary</h3>

this pr adds runtime support for `gpt-5.4` and optional `gpt-5.4-pro` across model normalization, reasoning configuration, prompt-family detection, fallback policy, and documentation. the rollout is conservative: `gpt-5.4-pro` stays opt-in in shipped templates and a default fallback edge (`gpt-5.4-pro → gpt-5.4`) is wired into the existing `DEFAULT_UNSUPPORTED_CODEX_FALLBACK_CHAIN`. previously-flagged gaps (`gpt-5.4-pro-none` missing from `MODEL_MAP`, stripped `gpt-5.2-codex` normalization tests) have been addressed.

key changes:
- `model-map.ts` — 12 new entries for `gpt-5.4` and `gpt-5.4-pro` variants (including `gpt-5.4-pro-none`)
- `request-transformer.ts` — `gpt-5.4-pro` pattern checked before `gpt-5.4`; `none→low` upgrade for pro; `supportsXhigh` and `supportsNone` flags updated correctly
- `codex.ts` — `"gpt-5.4"` added as a `ModelFamily`; both `gpt-5.4` and `gpt-5.4-pro` resolve to the same family (`gpt_5_2_prompt.md` / `gpt-5.4-instructions.md`), which is intentional and test-confirmed; `prewarmCodexInstructions` now includes `gpt-5.4`
- `fetch-helpers.ts` — `gpt-5.4-pro → gpt-5.4` fallback edge added; entitlement error message updated
- `index.ts` — `gpt-5.4` added first in quota probe list
- config templates and all docs updated to default to `gpt-5.4`
- 1797 tests across 56 files passing; new dedicated files `test/gpt54-models.test.ts`, `test/edge-cases.test.ts`, `test/helper-utils.test.ts` added

the model normalization logic is correct, regex priority ordering is right (pro before base), xhigh/none/upgrade rules are consistent, fallback chain is clean, and test coverage is broad. one test description needs clarity for readability.

<h3>Confidence Score: 4/5</h3>

- safe to merge; model logic is correct with proper regex priority, fallback chain is clean, and test coverage is broad.
- model normalization logic is correct, pro/base distinction is properly prioritized in regex patterns, reasoning effort configurations are appropriate (none→low upgrade for pro, both support xhigh), and fallback behavior is well-tested. one test description needs rewording for clarity. no safety or functional concerns.
- test/gpt54-models.test.ts — test description naming (minor clarity issue)

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["incoming model string\n(e.g. openai/gpt-5.4-pro-high)"] --> B["strip provider prefix\n(split on '/')"]
    B --> C["MODEL_MAP lookup\ngetNormalizedModel()"]
    C -- hit --> D["normalized model ID\n(e.g. gpt-5.4-pro)"]
    C -- miss --> E["modelId.toLowerCase()"]
    E --> F{regex chain}
    F -- matches gpt-5.4-pro --> G["return 'gpt-5.4-pro'"]
    F -- matches gpt-5.4 --> H["return 'gpt-5.4'"]
    F -- matches other --> I["existing families\n(gpt-5.2, gpt-5-codex, etc.)"]
    F -- no match --> J["return 'gpt-5.1' (fallback)"]

    D --> K["getReasoningConfig()"]
    G --> K
    H --> K

    K --> L{isGpt54Pro?}
    L -- yes --> M["supportsNone = false\nnone → low upgrade\nsupportsXhigh = true"]
    L -- no --> N{isGpt54General?}
    N -- yes --> O["supportsNone = true\nsupportsXhigh = true"]

    K --> P["getModelFamily() for prompts"]
    P --> Q["both gpt-5.4 and gpt-5.4-pro\nmap to 'gpt-5.4' family\n→ gpt_5_2_prompt.md\n→ gpt-5.4-instructions.md"]

    G --> R{fallback policy?}
    R -- fallback enabled --> S["DEFAULT fallback chain:\ngpt-5.4-pro → gpt-5.4"]
    R -- strict --> T["entitlement error returned"]
```

<a href="https://app.greptile.com/ide/codex?prompt=Fix%20the%20following%201%20code%20review%20issues.%20Work%20through%20them%20one%20at%20a%20time%2C%20proposing%20concise%20fixes.%0A%0A---%0A%0A%23%23%23%20Issue%201%20of%201%0Atest%2Fgpt54-models.test.ts%3A123-126%0Athe%20test%20title%20says%20%22should%20not%20confuse%20gpt-5.4-pro%20with%20base%20gpt-5.4%22%2C%20but%20the%20assertions%20verify%20they%20both%20map%20to%20the%20same%20family%20%28%60%22gpt-5.4%22%60%29.%20the%20wording%20implies%20they%20should%20differ%2C%20which%20is%20backwards%20from%20what%20the%20test%20checks.%0A%0Asuggest%20renaming%20to%20clarify%20the%20actual%20behavior%3A%0A%0A%60%60%60suggestion%0Ait%28%22gpt-5.4-pro%20belongs%20to%20the%20gpt-5.4%20model%20family%22%2C%20%28%29%20%3D%3E%20%7B%0A%20%20%20%20expect%28getModelFamily%28%22gpt-5.4-pro%22%29%29.toBe%28%22gpt-5.4%22%29%3B%0A%20%20%20%20expect%28getModelFamily%28%22gpt%205.4%20pro%22%29%29.toBe%28%22gpt-5.4%22%29%3B%0A%7D%29%3B%0A%60%60%60%0A%0A&repo=ndycode%2Foc-chatgpt-multi-auth"><picture><source media="(prefers-color-scheme: dark)" srcset="https://img.shields.io/badge/Fix_All_in_Codex-black?logo=data%3Aimage%2Fpng%3Bbase64%2CiVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAYAAABXAvmHAAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAMKADAAQAAAABAAAAMAAAAADbN2wMAAAGD0lEQVRoBe1ZC4hUVRg%2B%2Fx1H81GrjDNrarUmVpCVIokEFUrlo6SoRqFQBHfHfVFEglD0EqFQ1Mp9yO74AFuipQcZuRaEFhERtqgkheHW5hLN7I7rWu1jdub%2BfWfqjHfu3Lneuzuzu4F3Wc75n%2Bf%2F7vnPuf85I8TVZ3TfAOVj%2BE0bovO8Xs8qFmIJCQ7Ap45%2BRAg6q3PyWH048DV4YOX%2FGRaAitLORR7SdgsS99qFxsw%2FMtPLdWHf%2B3Z6Q5ENFQBVhbq2CaYtRMLjdGBMwdEBPbEhHC7G7OTncQ0gGGz2BKYuO0BE64YSArNoh91RQXwLsbiOieKC%2BXckWCuTfqSuMXDSjV%2FXAKrLOncJ0p5zM4gbXRbcKnSxtTY8%2FWMndq4AVIY6lxJrXyBtXNk5CcSsg3XTNEjxUEPDzF6zzEhrRuJKfU3QzpEIXsaBFH16PE84Ggyen2gXl2MAVaWxB%2BF2oZ2zvMuwuxVPm3TAzq8jAOUbuxYjabbbOSqgbG11KBbM5d82lytCnQ9rTK9gOu%2FO5WAk%2BNh%2Bfz1z9vS848eXJszjWQIIhU54vTxnD%2FJ9k9lg1GjWH61p9B82j5%2BVQjL48VxyZEwFj6iZtNXm4CU9zsz0ijk1yPcHzPxRp5nvsoohYwaqyqJrkFMhK8VR55GYaRVDGoAsEQR5XrdSMvNQDsi%2FC2Z%2BYWm6xsp%2FGkCgaNkjePs3WykZedgRBnVOLKhp9PmYk2uNskL2iTlrB5LjpQEg7x9zGMBP9eHi01K3tjHQjNlodmg3XLV2KwdpANh1llgpZPFYzK8siy5QfE70b0Y%2B2dYrSnc4LarW6ys3RGeYfaQBQHCTWWhFy1oIR4A3lKzuwKzzWA%2BO1o6yGUqL9L6BvNrhYPDMeKO9AQDZFk1GIyBYXl0WWa54bR3dO7A22hRdqFZWBIGiwLNG%2F5cBMLs6JbEYtzO1c8FbS8u8ARxKNhsdF6yvaS9WBqNTlP%2FLAAS1KqaTFrNwu3%2FqsnSpUds4%2FSOA%2BM6J7XB0kEpFWpEnXdwZAHCLW8ca0RajDdLoEyNdqD6TeFL5TgOIU%2FxdLMa4EjhpdSHCRj28nREpQXB1k66O0wBwdOtCQG8bA7Lts%2FilN35ph9JJLWqi%2BxVd0JbIv3LlzxPkGGkAkuhP6ltR952T%2FSs9OhbtwYNz%2BqWerGBRhrx5JZt8yvv6OpLSXwaA%2Ffv9fyZ1luWB5Wc7HQDzl3Xh6R8qepy46RkcN29TdKFbrLWYOtxkAJAD14f932M3idkGkby8ZcqvI%2BHUlkufda4WCX1WXFyc2tM3OCXS3TtJ%2FsfFwGRJS36in2cL1itSRWIuRxl8TpUykpV1Hkh96YiKM%2FQzCD5Xs99%2FQrHwddyOD8y1ija1p3C%2FU2viWZE9YO6tKovJbTldplgpSh5Kl0%2BVLGsGAhOLcgWTssEdZ0lFaSxVN1WXxZ5A8Dlv6DDQ3PL1XbPUYHZtaWlsNnaxEjsdKcN9Ub%2BIc5PSg03ms27dH5OLJnr%2FyuRmUqmpJu5EWSRvou0fFhcx7CkUY314dTo%2BgEhhy%2BcOcG%2B0lBiYGHtXbaPvecXKAiAFVaFYFwQ%2BpTRWWsxoa9v57ntSpct%2FQWWl0L98%2FmasBK3iwJtvF8n448bgpSwHAPGBMhwLLfL%2B7CAN3Fe7b2a7OR5LANHuvmYkapdZeaRpxDCI7XU3LnkXolL4zWp8yzUgFXGNXo5r9Horo8LzeEBn2hNn8VY47OuwGy8nAGmEffk97Bpr7BwUQoaUeQnl%2BTYnvi1TSBm2dVxYjy3wc0WPRIu06Rmknj1Ox7IFIFd8pDuyGk4zymanzoeih7f%2FQkPDXPlldvTYppDRg7yp9jC9hiP9IiM%2Fr30WLbhvWuXGp2MAymnqp1WNVqCkmI%2F0wjUHyVuCi7hXkovtITgsUbquWuZvu3tpeVOT75IbO9cA7JxXPNU%2BTZs8Za%2F7hc%2Fv9PQlyg8dmvG3nX8rWV4BqAEqN0ZXaJom022x4lm1WFtfoTx6ta7Bf8xK7oRXEABq4IrSyJ2keVbg9%2BBbkWpIN%2FZidiK64JN6MvnZ3n0zflC6V9v%2F6xv4B1Ee8WrvhIOWAAAAAElFTkSuQmCC&logoColor=white"><img alt="Fix All in Codex" src="https://img.shields.io/badge/Fix_All_in_Codex-white?logo=data%3Aimage%2Fpng%3Bbase64%2CiVBORw0KGgoAAAANSUhEUgAAADAAAAAwCAYAAABXAvmHAAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAMKADAAQAAAABAAAAMAAAAADbN2wMAAAGD0lEQVRoBe1ZC4hUVRg%2B%2Fx1H81GrjDNrarUmVpCVIokEFUrlo6SoRqFQBHfHfVFEglD0EqFQ1Mp9yO74AFuipQcZuRaEFhERtqgkheHW5hLN7I7rWu1jdub%2BfWfqjHfu3Lneuzuzu4F3Wc75n%2Bf%2F7vnPuf85I8TVZ3TfAOVj%2BE0bovO8Xs8qFmIJCQ7Ap45%2BRAg6q3PyWH048DV4YOX%2FGRaAitLORR7SdgsS99qFxsw%2FMtPLdWHf%2B3Z6Q5ENFQBVhbq2CaYtRMLjdGBMwdEBPbEhHC7G7OTncQ0gGGz2BKYuO0BE64YSArNoh91RQXwLsbiOieKC%2BXckWCuTfqSuMXDSjV%2FXAKrLOncJ0p5zM4gbXRbcKnSxtTY8%2FWMndq4AVIY6lxJrXyBtXNk5CcSsg3XTNEjxUEPDzF6zzEhrRuJKfU3QzpEIXsaBFH16PE84Ggyen2gXl2MAVaWxB%2BF2oZ2zvMuwuxVPm3TAzq8jAOUbuxYjabbbOSqgbG11KBbM5d82lytCnQ9rTK9gOu%2FO5WAk%2BNh%2Bfz1z9vS848eXJszjWQIIhU54vTxnD%2FJ9k9lg1GjWH61p9B82j5%2BVQjL48VxyZEwFj6iZtNXm4CU9zsz0ijk1yPcHzPxRp5nvsoohYwaqyqJrkFMhK8VR55GYaRVDGoAsEQR5XrdSMvNQDsi%2FC2Z%2BYWm6xsp%2FGkCgaNkjePs3WykZedgRBnVOLKhp9PmYk2uNskL2iTlrB5LjpQEg7x9zGMBP9eHi01K3tjHQjNlodmg3XLV2KwdpANh1llgpZPFYzK8siy5QfE70b0Y%2B2dYrSnc4LarW6ys3RGeYfaQBQHCTWWhFy1oIR4A3lKzuwKzzWA%2BO1o6yGUqL9L6BvNrhYPDMeKO9AQDZFk1GIyBYXl0WWa54bR3dO7A22hRdqFZWBIGiwLNG%2F5cBMLs6JbEYtzO1c8FbS8u8ARxKNhsdF6yvaS9WBqNTlP%2FLAAS1KqaTFrNwu3%2FqsnSpUds4%2FSOA%2BM6J7XB0kEpFWpEnXdwZAHCLW8ca0RajDdLoEyNdqD6TeFL5TgOIU%2FxdLMa4EjhpdSHCRj28nREpQXB1k66O0wBwdOtCQG8bA7Lts%2FilN35ph9JJLWqi%2BxVd0JbIv3LlzxPkGGkAkuhP6ltR952T%2FSs9OhbtwYNz%2BqWerGBRhrx5JZt8yvv6OpLSXwaA%2Ffv9fyZ1luWB5Wc7HQDzl3Xh6R8qepy46RkcN29TdKFbrLWYOtxkAJAD14f932M3idkGkby8ZcqvI%2BHUlkufda4WCX1WXFyc2tM3OCXS3TtJ%2FsfFwGRJS36in2cL1itSRWIuRxl8TpUykpV1Hkh96YiKM%2FQzCD5Xs99%2FQrHwddyOD8y1ija1p3C%2FU2viWZE9YO6tKovJbTldplgpSh5Kl0%2BVLGsGAhOLcgWTssEdZ0lFaSxVN1WXxZ5A8Dlv6DDQ3PL1XbPUYHZtaWlsNnaxEjsdKcN9Ub%2BIc5PSg03ms27dH5OLJnr%2FyuRmUqmpJu5EWSRvou0fFhcx7CkUY314dTo%2BgEhhy%2BcOcG%2B0lBiYGHtXbaPvecXKAiAFVaFYFwQ%2BpTRWWsxoa9v57ntSpct%2FQWWl0L98%2FmasBK3iwJtvF8n448bgpSwHAPGBMhwLLfL%2B7CAN3Fe7b2a7OR5LANHuvmYkapdZeaRpxDCI7XU3LnkXolL4zWp8yzUgFXGNXo5r9Horo8LzeEBn2hNn8VY47OuwGy8nAGmEffk97Bpr7BwUQoaUeQnl%2BTYnvi1TSBm2dVxYjy3wc0WPRIu06Rmknj1Ox7IFIFd8pDuyGk4zymanzoeih7f%2FQkPDXPlldvTYppDRg7yp9jC9hiP9IiM%2Fr30WLbhvWuXGp2MAymnqp1WNVqCkmI%2F0wjUHyVuCi7hXkovtITgsUbquWuZvu3tpeVOT75IbO9cA7JxXPNU%2BTZs8Za%2F7hc%2Fv9PQlyg8dmvG3nX8rWV4BqAEqN0ZXaJom022x4lm1WFtfoTx6ta7Bf8xK7oRXEABq4IrSyJ2keVbg9%2BBbkWpIN%2FZidiK64JN6MvnZ3n0zflC6V9v%2F6xv4B1Ee8WrvhIOWAAAAAElFTkSuQmCC&logoColor=black"></picture></a>

<sub>Last reviewed commit: bff8479</sub>

### CODE REVIEW

**Comment by ndycode** [comment]
@coderabbitai review

@greptile-ai please review this PR for runtime/model-routing correctness and test coverage completeness.

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,23 @@ all notable changes to this project. dates are ISO format (YYYY-MM-DD).
 - **doctor safe-fix edge path**: `codex-doctor fix` now reports a clear non-crashing message when no eligible account is available for auto-switch.
 - **first-time import flow**: `codex-import` no longer fails with `No accounts to export` when storage is empty; pre-import backup is skipped cleanly in zero-account setups.
 
+## [5.4.2] - 2026-03-06
```
[the remark this answers is not in the document]
Comment (ndycode) [reply CHANGELOG.md]: System.Collections.Hashtable.body

File: docs/development/TESTING.md [hunk 2]
```diff
@@ -605,35 +605,45 @@ opencode
...
-normalizeModel("my-codex-model")        // → "gpt-5.1-codex" ✅
+normalizeModel("my-codex-model")        // → "gpt-5-codex" ✅
 normalizeModel("gpt-5")                 // → "gpt-5.1" ✅
 normalizeModel("gpt-5-mini")            // → "gpt-5.1" ✅
 normalizeModel("gpt-5-nano")            // → "gpt-5.1" ✅
-normalizeModel("GPT 5 High")            // → "gpt-5.1" ✅
-normalizeModel(undefined)               // → "gpt-5.1" ✅
+normalizeModel("GPT 5.4 Pro High")      // → "gpt-5.4-pro" ✅
+normalizeModel(undefined)                 // → "gpt-5.1" ✅
 normalizeModel("random-model")          // → "gpt-5.1" ✅ (fallback)
```
[the remark this answers is not in the document]
Comment (ndycode) [reply docs/development/TESTING.md:625]: System.Collections.Hashtable.body

File: test/request-transformer.test.ts [hunk 3]
```diff
@@ -83,13 +83,23 @@ describe('Request Transformer Module', () => {
...
+					expect(normalizeModel('gpt-5.4-medium')).toBe('gpt-5.4');
+					expect(normalizeModel('gpt-5.4-high')).toBe('gpt-5.4');
+					expect(normalizeModel('gpt-5.4-xhigh')).toBe('gpt-5.4');
+					expect(normalizeModel('openai/gpt-5.4-xhigh')).toBe('gpt-5.4');
+				});
+
+				it('should normalize gpt-5.4-pro presets', async () => {
+					expect(normalizeModel('gpt-5.4-pro')).toBe('gpt-5.4-pro');
+					expect(normalizeModel('gpt-5.4-pro-low')).toBe('gpt-5.4-pro');
+					expect(normalizeModel('gpt-5.4-pro-medium')).toBe('gpt-5.4-pro');
```
[the remark this answers is not in the document]
Comment (ndycode) [reply test/request-transformer.test.ts:108]: System.Collections.Hashtable.body

File: lib/request/helpers/model-map.ts [hunk 4]
```diff
@@ -58,6 +58,25 @@ export const MODEL_MAP: Record<string, string> = {
...
+
+	// ============================================================================
+	// GPT-5.4 Pro Models (optional/manual config)
+	// ============================================================================
+	"gpt-5.4-pro": "gpt-5.4-pro",
+	"gpt-5.4-pro-low": "gpt-5.4-pro",
+	"gpt-5.4-pro-medium": "gpt-5.4-pro",
+	"gpt-5.4-pro-high": "gpt-5.4-pro",
+	"gpt-5.4-pro-xhigh": "gpt-5.4-pro",
+
```
[the remark this answers is not in the document]
Comment (ndycode) [reply lib/request/helpers/model-map.ts:80]: System.Collections.Hashtable.body

File: test/request-transformer.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ndycode) [reply test/request-transformer.test.ts:108]: Fixed in beb88bd: restored additive coverage for gpt-5.2-codex normalization presets in test/request-transformer.test.ts while keeping the new GPT-5.4 tests.

File: lib/request/helpers/model-map.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (ndycode) [reply lib/request/helpers/model-map.ts:80]: Fixed in beb88bd: added missing gpt-5.4-pro-none -> gpt-5.4-pro mapping in MODEL_MAP and added assertions in test/model-map.test.ts and test/request-transformer.test.ts.

File: docs/development/TESTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ndycode) [reply docs/development/TESTING.md:625]: Fixed in beb88bd: aligned fallback-related normalizeModel examples in docs/development/TESTING.md to gpt-5.1 and updated stale unit-test snippets to match current normalization behavior.

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ndycode) [reply CHANGELOG.md]: Fixed in beb88bd: changed changelog header to ## [5.4.2] - 2026-03-05 to avoid future-dated release metadata.

**Comment by ndycode** [comment]
Addressed all review findings in commit `beb88bd`.

Also fixed the outside-diff review note by tightening the property test assertion in `test/property/transformer.property.test.ts` from `not.toBe("none")` to `toBe("low")` for the codex/pro `none -> low` contract.

Validation rerun after fixes:
- `npm run typecheck`
- `npm run lint`
- `npm test` (56 files, 1798 tests)

**Comment by ndycode** [comment]
@greptile-ai please re-review latest commit eb88bd.

@coderabbitai review

**Comment by ndycode** [comment]
Followed up on review `#3900326193` in commit `d9f8617`.

Changes applied:
- `docs/configuration.md`: changed the "look for:" fence from ``` to ```text to satisfy MD040.
- `test/index.test.ts`: set `getFallbackToGpt52OnUnsupportedGpt53` mock to `false` in the GPT-5.4-pro fallback test to isolate that behavior from legacy GPT-5.3 fallback toggles.

Validation after this commit:
- `npm run typecheck`
- `npm run lint`
- `npm test` (56 files, 1798 tests)

**Comment by ndycode** [comment]
@coderabbitai review

@greptile-ai please re-review latest commit d9f8617.

**Comment by ndycode** [comment]
Addressed remaining review issues in `bff8479` with targeted hardening and regression coverage.

### What was fixed
- Replaced broad substring checks with boundary-aware matching for GPT-5.4 family detection:
  - `lib/request/request-transformer.ts`
  - `lib/prompts/codex.ts`
- Clarified docs for reasoning effort coercion:
  - `docs/configuration.md` now explicitly states `none -> low` applies to codex/pro families (including `gpt-5.4-pro`).
- Expanded regression tests to close model-id edge cases and map coverage:
  - `test/gpt54-models.test.ts`
    - Added explicit `gpt-5.4-pro-none` assertions.
    - Tightened pro-variant count expectation to include `none`.
    - Updated edge-case expectation so `gpt-5.40`/`gpt-5.44` do **not** match `gpt-5.4`.
  - `test/request-transformer.test.ts`
    - Added checks that `gpt-5.40` and `gpt-5.4pro` do not normalize to `gpt-5.4`.
  - `test/codex-prompts.test.ts`
    - Added check that `gpt-5.40` is not classified as `gpt-5.4` family.

### Validation
- `npm run typecheck` ✅
- `npm run lint` ✅
- `npm test` ✅ (59 files, 1935 tests)

@coderabbitai review
@greptile-ai please re-review latest commit `bff8479`
